pkgs/top-level/config.nix

```diff
@@ -36,16 +55,326 @@ let
       internal = true;
     };
 
+    unknowns = mkOption {
+      type = types.attrsOf (types.uniq types.unspecified);
+      default = {};
+      internal = true;
+    };
+
+    /* System options */
+
+    platform = mkOption {
+      type = types.uniq types.unspecified;
+      description = "Local platform override";
+    };
+
+    localSystem = mkOption {
+      type = types.attrs;
+      # Allow setting the platform in the config file. This take precedence over
+      # the inferred platform, but not over an explicitly passed-in one.
+      apply = x: lib.systems.elaborate
+        (x // lib.optionalAttrs options.platform.isDefined config.platform);
+      description = "Local system";
+    };
+
+    # FIXME! NOTE: Not all functions are congruent in Nix, @oxij really hopes this is just a bug
+    # > $ nix-instantiate --eval -E 'with import ./lib; let f = systems.elaborate; x = { system = "x86_64-linux"; }; in f x == f x'
+    # > false
+    # This means that we have to hack around this bug to assign (!) `crossSystem` to `config.localSystem` by default
+    # else `crossSystem != localSystem` check in ../stdenv/default.nix will always be `true`,
+    # which would in turn cause stdenv to try to cross compile to our own platform, stdenv was not made for that.
+    # Hence we have to use the `apply` implementation below.
+    crossSystem = mkOption {
+      type = types.nullOr types.attrs;
+      apply = x: if x == null then config.localSystem else lib.systems.elaborate x;
+      description = "Cross system";
+    };
+
     /* Config options */
 
+    inHydra = mkMeta {
+      internal = true;
+      description = ''
+        If we're in hydra, we can dispense with the more verbose error
+        messages and make problems easier to spot.
+      '';
+    };
+
+    allowBroken = mkMeta {
+      default = builtins.getEnv "NIXPKGS_ALLOW_BROKEN" == "1";
+      defaultText = ''builtins.getEnv "NIXPKGS_ALLOW_BROKEN" == "1"'';
+      feature = "permit evaluation of broken packages";
+    };
```

Comment:
We have some of these in `pkgs/top-level/config.nix` now, but they're not nearly as complete, so they're still worth adding.

Reply:
(except for the new ones that don't exist yet)